src/atomate2/aims/flows/gw.py

```diff
@@ -0,0 +1,91 @@
+"""GW workflows for FHI-aims with automatic convergence."""
+from copy import deepcopy
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Union
+
+from jobflow import Flow
+from pymatgen.core import Molecule, Structure
+
+from atomate2.aims.jobs.base import BaseAimsMaker, ConvergenceMaker
+from atomate2.aims.jobs.core import BandStructureMaker, GWMaker, StaticMaker
+from atomate2.aims.sets.bs import BandStructureSetGenerator, GWSetGenerator
+from atomate2.aims.sets.core import StaticSetGenerator
+from atomate2.aims.utils.msonable_atoms import MSONableAtoms
+
+
+@dataclass
+class PeriodicGWConvergenceMaker(BaseAimsMaker):
```

Comment:
I'm a bit conflicted by this class. This subclasses the `BaseAimsMaker` which is a `job` maker, but then this class itself is a `Flow` maker. So I think this will result in confusion. I thing the reason you've done it like this is to only have one instance of the `input_set_generator.parameters` that can be shared between all subjobs. However, I don't think this is the best way to do it.

My suggestion would be:
1. Don't subclass `BaseAimsMaker`.
2. Add two fields to the dataclass `static_maker` and `gw_maker`. These can have default values and use these makers inside the `make` function.
3. If you need to update the parameters for the convergence, then you can just use a powerup or the `update_maker_kwargs` function. This will update all parameters in the flow simultaneously.
4. If you want to support both molecules and crystals, I would make two subclasses of this maker, each with different defaults for the `static_maker`. Alternatively, you can add support inside one class, but the static will need to be added dynamically. See my previous comment about using `Response(replace=flow)`.

Reply:
I also didn't like it, it's there just because I was specifically asked for it. Ended up subclassing it from the `ConvergenceMaker` class.

Reply:
I don't have an issue with the concept of the class, just how it has been implemented. I think all the issues can be resolved following the steps I outlined above.